lib/generators/shopify_app/add_extension/add_extension_generator.rb

```diff
@@ -0,0 +1,65 @@
+require 'rails/generators/base'
+
+module ShopifyApp
+  module Generators
+    class AddExtensionGenerator < Rails::Generators::Base
+      source_root File.expand_path('../templates', __FILE__)
+
+      class_option :type, type: :string, aliases: "-t", required: true
+
+      def generate_app_extension
+        return unless extension_type_valid?
```

Comment:
The design of this generator in general is a tad design-breaking since really "add_extension" translates to "add_some_controllers_and_jobs_and_maybe_some_graphql_stuff" and currently every controller generator gets it's own folder. 

I chose not to follow that since the hmac generation and routes injection code will stay the same across extensions and probably some more code to follow. 

I'm also ok with just naming this `add_marketing_activities_extension` and not handling any cases of multiple extensions until we get there.

Reply:
gonna go with this^

Reply:
renaming would be good as we can just make a parent coordinator task that calls many children tasks.

Reply:
Would the way it is now be fine?